Classes/Utility/AttributeUtility.php

```diff
@@ -158,34 +210,56 @@ public static function findAttributesForAttributeSet(int $attributeSetId): array
     }
 
     /**
-     * Find a specific attribute value for a product.
+     * Find all attribute values for product with attribute identifiers as keys.
      *
      * @param int $productId
-     * @param int $attributeId
+     * @param array $attributeIds
      * @param string $selectFields
      * @return array|null
+     * @throws \Doctrine\DBAL\Driver\Exception
      */
-    public static function findAttributeValue(int $productId, int $attributeId, string $selectFields = '*'): ?array
+    public static function findAttributeValues(int $productId, array $attributeIds, string $selectFields = '*'): ?array
     {
         $queryBuilder = self::getQueryBuilderForTable(AttributeValueRepository::TABLE_NAME);
+        $queryBuilder->getRestrictions()->removeAll();
 
-        $row = $queryBuilder
+        $attributeValues = $queryBuilder
             ->select(...GeneralUtility::trimExplode(',', $selectFields, true))
-            ->from(AttributeValueRepository::TABLE_NAME)
+            ->from(AttributeRepository::TABLE_NAME)
+            ->join(
+                AttributeRepository::TABLE_NAME,
+                AttributeValueRepository::TABLE_NAME,
+                'av',
+                $queryBuilder->expr()->eq('av.attribute', AttributeRepository::TABLE_NAME . '.uid')
+            )
             ->where(
                 $queryBuilder->expr()->eq('product', $queryBuilder->createNamedParameter($productId)),
-                $queryBuilder->expr()->eq('attribute', $queryBuilder->createNamedParameter($attributeId))
+                $queryBuilder->expr()->in('attribute', $attributeIds)
             )
             ->execute()
-            ->fetchAssociative();
+            ->fetchAllAssociative();
 
-        if (is_array($row)) {
-            return $row;
+        if (is_array($attributeValues)) {
+            return array_column($attributeValues, null, 'identifier');
         }
 
         return null;
     }
 
+    /**
+     * Find attribute value for product with given attribute id.
+     *
+     * @param int $productId
+     * @param int $attributeId
+     * @param string $selectFields
+     * @return array|null
+     * @throws \Doctrine\DBAL\Driver\Exception
+     */
+    public static function findAttributeValue(int $productId, int $attributeId, string $selectFields = '*'): ?array
```

Comment:
To reflect reality. ;-)

```suggestion
     * @return array
     * @throws \Doctrine\DBAL\Driver\Exception
     */
    public static function findAttributeValue(int $productId, int $attributeId, string $selectFields = '*'): array
```